### PR TITLE
search updates

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-1606
Figma link (if any):

## What changed (plus any additional context for devs)

pasta of some changes in app

### CODE REVIEW

**Review by derHowie — APPROVED** [review]
lgtm

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
See an upstream crash related the failed e2e test in the Cmd K suite. Trigger Cmd K -> Paste contract address for unowned token `0xdac17f958d2ee523a2206206994597c13d831ec7` -> Crash

> popup.js:239364 Uncaught TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at popup.js:239364:47
    at Array.map (<anonymous>)
    at getTokenCommandRelevance (popup.js:239364:30)
    at calculateCommandRelevance (popup.js:239435:20)
    at popup.js:239477:27
    at Array.forEach (<anonymous>)
    at popup.js:239473:29
    at popup.js:236572:78
    at updateMemo (popup.js:180541:19)
    at Object.useMemo (popup.js:181181:16)
popup.js:239364 Uncaught TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at popup.js:239364:47
    at Array.map (<anonymous>)
    at getTokenCommandRelevance (popup.js:239364:30)
    at calculateCommandRelevance (popup.js:239435:20)
    at popup.js:239477:27
    at Array.forEach (<anonymous>)
    at popup.js:239473:29
    at popup.js:236572:78
    at updateMemo (popup.js:180541:19)
    at Object.useMemo (popup.js:181181:16)
> popup.js:181982 The above error occurred in the <CommandK> component:

    at CommandK (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:236537:143)
    at div
    at Box (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:225994:82)
    at div
    at ColorContextProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:226095:33)
    at Box (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:225994:82)
    at div
    at Box (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:225994:82)
    at FullScreenBackground (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:240815:33)
    at RootLayout (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:234133:95)
    at RenderedRoute (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:197817:5)
    at RenderErrorBoundary (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:197763:5)
    at Routes (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:198264:5)
    at Router (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:198196:15)
    at RouterProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:198049:5)
    at Routes
    at div
    at ColorContextProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:226095:33)
    at Box (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:225994:82)
    at AuthProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:248677:25)
    at div
    at ThemeProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:226143:26)
    at QueryClientProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:317045:3)
    at QueryClientProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:317045:3)
    at PersistQueryClientProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:316974:3)
    at Hydrate (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:377703:13)
    at WagmiProvider (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:377443:13)
    at App (chrome-extension://eaainhkbimoimhlhlidlnfamifeolpjg/popup.js:233524:122)

> React will try to recreate this component tree from scratch using the error boundary you provided, RenderErrorBoundary.
logCapturedError @ popup.js:181982
popup.js:197803 React Router caught the following error during render TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at popup.js:239364:47
    at Array.map (<anonymous>)
    at getTokenCommandRelevance (popup.js:239364:30)
    at calculateCommandRelevance (popup.js:239435:20)
    at popup.js:239477:27
    at Array.forEach (<anonymous>)
    at popup.js:239473:29
    at popup.js:236572:78
    at updateMemo (popup.js:180541:19)
    at Object.useMemo (popup.js:181181:16) Object
componentDidCatch @ popup.js:197803

**Comment by DanielSinclair** [comment]
Seeing a similar error with the same test case: `Trigger Cmd K -> Paste contract address for unowned token 0xdac17f958d2ee523a2206206994597c13d831ec7 -> Crash`
> React Router caught the following error during render TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at utils.ts:75:47
    at Array.map (<anonymous>)
    at utils.ts:75:30
    at lx (utils.ts:146:20)
    at utils.ts:193:27
    at Array.forEach (<anonymous>)
    at i (utils.ts:189:29)
    at CommandK.tsx:56:16
    at Object.Ja [as useMemo] (react-dom.production.min.js:179:119)
    at t.useMemo (react.production.min.js:25:208)
    at sy (CommandK.tsx:55:30)
    at _a (react-dom.production.min.js:167:137)
    at Ts (react-dom.production.min.js:197:258)
    at _c (react-dom.production.min.js:292:88)
    at wu (react-dom.production.min.js:280:389)
    at bu (react-dom.production.min.js:280:320)
    at yu (react-dom.production.min.js:280:180)
    at au (react-dom.production.min.js:271:88)
    at iu (react-dom.production.min.js:268:429)
    at x (scheduler.production.min.js:13:203)
    at MessagePort.k (scheduler.production.min.js:14:128)

File: src/core/resources/search/tokenSearch.ts [hunk 1]
```diff
@@ -178,6 +193,7 @@ export function useTokenSearch(
     }),
     queryFn: tokenSearchQueryFunction,
     ...config,
+    placeholderData: (previousData) => previousData,
```
Comment (benisgold) [inline src/core/resources/search/tokenSearch.ts:196]: this is the default behavior so this does not do anything

**Review by benisgold — COMMENTED** [review]
can you pinpoint the change that fixes BX-1606? i can't find code related to this lol

File: src/core/resources/search/tokenSearch.ts (hunk 1 above)
Comment (greg-schrammel) [reply src/core/resources/search/tokenSearch.ts:196]: BX is on react query v5, `placeholderData: (previousData) => previousData,` is the same as the old `keepPreviousData`
https://tanstack.com/query/v5/docs/framework/react/guides/migrating-to-v5#removed-keeppreviousdata-in-favor-of-placeholderdata-identity-function

**Comment by derHowie** [comment]
@DanielSinclair your crash is not occurring in the branch, however I saw a different issue while attempting to repro: https://www.loom.com/share/4d0c018d80384a07b9347021e2549c13

I'm guessing from the description that these changes were meant to port over some behavior or performance improvements from the app, but I'm unsure what to test for.

I may need some more context on the purpose of these changes before I can fix the bug in the video and make sure this is good to go.

**Comment by DanielSinclair** [comment]
@derHowie Wasn't totally sure of the context either. I looked at the merge conflicts on Friday, but there are quite a few following Popular in Rainbow

**Comment by DanielSinclair** [comment]
Go for it 🫡

**Pull request closed without merging**